rai/node/node.cpp

```diff
@@ -428,21 +429,24 @@ void rai::network::receive_action (boost::system::error_code const & error, size
     {
         if (!rai::reserved_address (remote) && remote != endpoint ())
         {
-            network_message_visitor visitor (node, remote);
-            rai::message_parser parser (visitor, node.work);
-            parser.deserialize_buffer (buffer.data (), size_a);
-            if (parser.error)
+            if (!packet_filter || !packet_filter (buffer.data (), size_a))
```

Comment:
This is only used by the test.

Reply:
It seems crufty but ifdef seems cruftier.  Please give it a name that stands out like `_TESTING_packet_filter` so no one mistakes it for normal functionality.